lib/simulation/simulated_characteristic.dart

```diff
@@ -33,9 +33,9 @@ class SimulatedCharacteristic {
 
   Future<Uint8List> read() async => _value;
 
-  Future<void> write(Uint8List value) async {
+  Future<void> write(Uint8List value, {bool sendNotification = true}) async {
```

Comment:
Just a proposition. What about decoupling these two operations? I mean writing and setting value. In core BT android stack there is an option to set characteristic value locally (in cache) without performing actual writing operation. It can be achieved by simply calling characteristic.setValue().
Maybe we could do this the same way here?

Reply:
This doesn't really pertain to BLEmulator, but more to FlutterBleLib and its API. I don't see the benefit to be honest, but I'm not an expert.